File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/SupplyDelivery.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Timing } from "./Timing";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type SupplyDeliverySuppliedItem = BackboneElement & {
    item?: Reference | CodeableConcept;
    quantity?: Quantity;
};

export type SupplyDelivery = DomainResource & {
    patient?: Reference;
    supplier?: Reference;
    suppliedItem?: SupplyDeliverySuppliedItem;
    type?: CodeableConcept;
    status?: string;
    identifier?: Identifier[];
    basedOn?: Reference[];
    partOf?: Reference[];
    receiver?: Reference[];
    destination?: Reference;
    occurrence?: Timing | Period | string;
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/SupplyDelivery.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import Timing
from base import Quantity
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class SupplyDeliverySuppliedItem(BackboneElement):
    quantity: Optional[Quantity] = None
    item_reference: Optional[Reference] = None
    item_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class SupplyDelivery(DomainResource):
    patient: Optional[Reference] = None
    supplier: Optional[Reference] = None
    supplied_item: Optional[SupplyDeliverySuppliedItem] = None
    type: Optional[CodeableConcept] = None
    occurrence_timing: Optional[Timing] = None
    occurrence_period: Optional[Period] = None
    status: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    based_on: Optional[List[Reference]] = field(default_factory=list)
    part_of: Optional[List[Reference]] = field(default_factory=list)
    occurrence_date_time: Optional[str] = None
    receiver: Optional[List[Reference]] = field(default_factory=list)
    destination: Optional[Reference] = None